### PR TITLE
Tooltips sample integration

### PR DESCRIPTION
**Pull request by Velenir**
Adds

- Hook for Popper.js
- Hook with reasonable defaults ready to plug-n-play
- TooltipComponent for easy plug-n-play tooltips
- Sample tooltips on `/trade` page (style pending)

localhost_8080_trade_DAI-USDC_sell=0 buy=0 expires=2880

Tooltips don't have to look like that or like anything really, can be whole components inside

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr566--dexreact.review.gnosisdev.com

File: src/components/TradeWidget/TokenRow.tsx [hunk 1]
```diff
@@ -305,36 +307,42 @@ const TokenRow: React.FC<Props> = ({
       <div>
         <strong>{selectLabel}</strong>
         <span>
-          <button>+ Deposit</button>
+          <TooltipWrapper as="button" tooltip="Deposit">
+            + Deposit
+          </TooltipWrapper>
+          {/* <button>+ Deposit</button> */}
```
Comment (alfetopito) [inline src/components/TradeWidget/TokenRow.tsx]: dead code?

File: src/components/Tooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+import { State, Placement } from '@popperjs/core'
+import styled from 'styled-components'
+import { isElement, isFragment } from 'react-is'
+
+// visibility necessary for correct boundingRect calculation by popper
+const TooltipOuter = styled.div<Pick<TooltipBaseProps, 'isShown'>>`
+  visibility: ${(props): 'hidden' | false => !props.isShown && 'hidden'};
+`
+// can style anything but TOOLTIP_OFFSET fields, position and transform: rotate
+const TooltipArrow = styled.div`
```
Comment (W3stside) [inline src/components/Tooltip.tsx]: why all this styling? I thought the library handled styling more gracefully via props? or am i wrong

File: src/components/Tooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+const Wrapper: React.FC<WrapperProps> = ({ children, tooltip, placement, as }) => {
+  const { targetProps, tooltipProps } = usePopperDefault<HTMLDivElement>(placement)
+
+  const chilrenNumber = React.Children.count(children)
+
+  // can attach ref to element
+  // if children is a single element, not just text, and `as` tag|Component not specified
+  if (chilrenNumber === 1 && typeof as === 'undefined' && isElement(children) && !isFragment(children)) {
```
Comment (W3stside) [inline src/components/Tooltip.tsx]: what is this for? seems like crazy amount of adding of code

**Comment by W3stside** [comment]
not saying it's a bad thing, but there's a lot seemingly going on in this PR that seems to be adding to the library. The hook looks very complicated - what was the reasoning behind adding it vs using the library directly?

File: src/components/TradeWidget/TokenRow.tsx (hunk 1 above)
Comment (Velenir) [reply src/components/TradeWidget/TokenRow.tsx]: Not dead code. Didn't delete as `<TooltipWrapper>` is just an example. Maybe we don't need a tooltip for that button

File: src/components/Portal.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import { createPortal } from 'react-dom'
+
+// mounts children in the body
```
Comment (anxolin) [inline src/components/Portal.tsx]: Great that you added this comments

File: src/hooks/usePopper.ts [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+const defaultConfig: Partial<Options> & Pick<Options, 'modifiers'> = {
+  // default tooltip placement
+  placement: 'top',
+  modifiers: [
+    {
+      // move slightly to make space for the arrow element
+      name: 'offset',
+      options: {
+        offset: [0, 8],
```
Comment (anxolin) [inline src/hooks/usePopper.ts]: Is this 8 supposed to be the constant above?

File: src/components/Tooltip.tsx (hunk 2 above)
Comment (Velenir) [reply src/components/Tooltip.tsx]: It handle styling via attributes, like vanilla JS would. And even then only positioning through transforms, not actual style styling

File: src/components/Tooltip.tsx (hunk 3 above)
Comment (Velenir) [reply src/components/Tooltip.tsx]: To be able to clone Element later and pass it a `ref`


> not saying it's a bad thing, but there's a lot seemingly going on in this PR that seems to be adding to the library. The hook looks very complicated - what was the reasoning behind adding it vs using the library directly?

Library directly do anything with React.

File: src/hooks/usePopper.ts [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    fn: ({ state }) => {
+      // popper.state is mutable, need a new instance
+      setState({ ...state })
+    },
+  })
+
+  return finalConfig
+}
+
+interface PopperHookResult<T extends HTMLElement, U extends HTMLElement = HTMLDivElement> {
```
Comment (anxolin) [inline src/hooks/usePopper.ts]: Do we have a convention  for this interfaces? 
Personally I think it's fine to call it just Result, specially if you don't export it

**Comment by Velenir** [comment]
> not saying it's a bad thing, but there's a lot seemingly going on in this PR that seems to be adding to the library. The hook looks very complicated - what was the reasoning behind adding it vs using the library directly?

Library directly doesn't do anything with React.

File: src/hooks/usePopper.ts (hunk 5 above)
Comment (Velenir) [reply src/hooks/usePopper.ts]: Forgot to inckude, it's `TOOLTIP_OFFSET`

File: src/hooks/usePopper.ts (hunk 6 above)
Comment (Velenir) [reply src/hooks/usePopper.ts]: Renamed

**Pull request merged**